src/nidigital/metadata/enums_addon.py

```diff
@@ -2,4 +2,24 @@
 # Any changes to the API should be made here. enums.py is code generated
 
 enums_override_metadata = {
+    'PinState': {
+        '__str__':"""    if self.name == 'ZERO':
+            return str(0)
+        elif self.name == 'ONE':
+            return str(1)
+        elif self.name == 'NOT_A_PIN_STATE':
+            return "Not a Pin State"
+        elif self.name == 'PIN_STATE_NOT_ACQUIRED':
+            return "Pin State Not Acquired"
+        else:
+            return str(self.name)\n"""
+    },
+    'WriteStaticPinState': {
+        '__str__':"""    if self.name == 'ZERO':
+            return str(0)
+        elif self.name == 'ONE':
+            return str(1)
+        else:
+            return str(self.name)\n"""
+    }
```

Comment:
I think it would be more consistent with the existing metadata and codegen design, if the `__str__` is codegen'd based on  metadata for the enum values.

So, add `pretty_name` key under enum values key:
```
    'PinState': {
        'values': [
            {
                'name': 'NIDIGITAL_VAL_0',
                'python_name': 'ZERO',
                'pretty_name': '0',
                'value': 0
            },
```

Then the code generator would add the `__str__` method if any value has the `pretty_name` key.

@marcoskirsch, can you please share your opinion?

Reply:
I agree. Having nested if/else nested logic in the metada feels wrong and is inconsistent.

Reply:
Key off new field pretty_name in the code generator instead.